### DISCUSSION TITLE
Need Help with new handle()

### DISCUSSION THREAD

**Discussion by karlmira** [danielsaidi/RichTextKit#156] [Q&A] [answered]
Sorry for asking, but I'm not that familiar with Swift. 
`context.pasteText()` has changed to `handle(.pasteText())` but I don't know, how to continue.

Can someone give some help with this, please.

`context.pasteText(baustein.text ?? "", at: context.selectedRange.location, moveCursorToPastedContent: true)` needs to be replaced.

Thanks so far.

**Comment by danielsaidi** [comment]
Hi @karlmira 

The new way should be to trigger the action:

```swift
context.handle(
    .pasteText(
        .init(content: "",  at: context.selectedRange.location, moveCursor: true
    )
)
```

@DominikBucher12 Please correct me if I'm wrong.

**Reply by karlmira** [reply]
@danielsaidi 

thanks for your help, but this leads to an error

`'RichTextInsertion<T>' initializer is inaccessible due to 'internal' protection level`

**Reply by danielsaidi** [reply]
Oh, so it was. I've fixed it and added new convenience builders for the insertion. You find them in `main`.

**Comment by karlmira** [comment, marked as answer]
Here, how it works:

`
context.handle(
     .pasteText(
        .init(
			content: baustein.text ?? "",
			index: context.selectedRange.location,
			moveCursor: true
		)
	)
)
`

**Comment by danielsaidi** [comment]
I think I'll add a paste function to the context as well, as a shorthand. Then, you'll be able to do `context.paste(...)`. The insert approach is nice, but it's a bit of a pain to use it.
